precise/iaf_chxk_2008.cpp

```diff
@@ -410,35 +409,36 @@ nest::iaf_chxk_2008::update( Time const& origin, const long from, const long to
       }
     }
     // neuron should spike on threshold crossing only.
-    if ( vm_prev < P_.V_th && S_.y[ State_::V_M ] >= P_.V_th )
+    if ( vm_prev < P_.V_th and S_.y[ State_::V_M ] >= P_.V_th )
     {
-      // neuron is not absolute refractory
+      // Find precise spike time by linear interpolation.
+      // dt is time from spike to end of time step.
+      const double dt = B_.step_ * ( S_.y[ State_::V_M ] - P_.V_th ) / ( S_.y[ State_::V_M ] - vm_prev );
 
-      // Find precise spike time using linear interpolation
-      double sigma = ( S_.y[ State_::V_M ] - P_.V_th ) * B_.step_ / ( S_.y[ State_::V_M ] - vm_prev );
+      // Evolve AHP conductance from time of spike to end of step
+      const double delta_dg = V_.PSConInit_AHP * std::exp( -dt / P_.tau_ahp );
+      const double delta_g = delta_dg * dt;
 
-      double alpha = exp( -sigma / P_.tau_ahp );
-
-      double delta_g_ahp = V_.PSConInit_AHP * sigma * alpha;
-      double delta_dg_ahp = V_.PSConInit_AHP * alpha;
-
-      if ( P_.ahp_bug == true )
+      if ( P_.ahp_bug )
       {
-        // Bug in original code ignores AHP conductance from previous spikes
-        S_.y[ State_::G_AHP ] = delta_g_ahp;
-        S_.y[ State_::DG_AHP ] = delta_dg_ahp;
+        // Bug in original Fortran implementation by Casti et al ignores
+        // AHP conductance from previous spikes: at any time, only the
+        // alpha function "kicked off" by the most recent spike is included
```

Comment:
Could we change `et al` to `et al.`? This is the style we use for in-text citations throughout our documentation.

Reply:
Done!